tests/integration/pilot/sidecartls/main_test.go

```diff
@@ -0,0 +1,356 @@
+	})
+
+	config := modifyConfig(Config{
+		AppNamespace: appNamespace.Name(),
+	})
+
+	// Apply all configs
+	createConfig(t, g, config, SidecarConfig, appNamespace)
+	createConfig(t, g, config, AppConfig, appNamespace)
+	createConfig(t, g, config, IncludedConfig, appNamespace)
+	createConfig(t, g, config, StricteMtls, appNamespace)
+
+	time.Sleep(time.Second * 2)
+
+	nodeID := &model.Proxy{
+		ClusterID:   "integration-test",
+		ID:          fmt.Sprintf("httpbin.%s", appNamespace.Name()),
+		DNSDomain:   appNamespace.Name() + ".cluster.local",
+		Type:        model.SidecarProxy,
+		IPAddresses: []string{"1.1.1.1"},
+		Metadata: &model.NodeMetadata{
+			Namespace: appNamespace.Name(),
+		},
+		ConfigNamespace: appNamespace.Name(),
+	}
+	return p, nodeID, func() {
+		// allow disable mtls later
+		deleteConfig(t, g, config, StricteMtls, appNamespace)
+		createConfig(t, g, config, DisableMTLS, appNamespace)
+	}
+}
+
+func createConfig(t *testing.T, g galley.Instance, config Config, yaml string, namespace namespace.Instance) {
+	tmpl, err := template.New("Config").Parse(yaml)
+	if err != nil {
+		t.Errorf("failed to create template: %v", err)
+	}
+	var buf bytes.Buffer
+	if err := tmpl.Execute(&buf, config); err != nil {
+		t.Errorf("failed to create template: %v", err)
+	}
+	if err := g.ApplyConfig(namespace, buf.String()); err != nil {
+		t.Fatalf("failed to apply config: %v. Config: %v", err, buf.String())
+	}
+}
+
+func deleteConfig(t *testing.T, g galley.Instance, config Config, yaml string, namespace namespace.Instance) {
+	tmpl, err := template.New("Config").Parse(yaml)
+	if err != nil {
+		t.Errorf("failed to create template: %v", err)
+	}
+	var buf bytes.Buffer
+	if err := tmpl.Execute(&buf, config); err != nil {
+		t.Errorf("failed to create template: %v", err)
+	}
+	if err := g.DeleteConfig(namespace, buf.String()); err != nil {
+		t.Fatalf("failed to delete config: %v. Config: %v", err, buf.String())
+	}
+}
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite("sidecar_custom_tls_test", m).
+		RequireEnvironment(environment.Native).
+		Run()
+}
+
+func TestSidecarCustomTLS(t *testing.T) {
+	framework.Run(t, func(ctx framework.TestContext) {
+		configFn := func(c Config) Config {
+			return c
+		}
+		p, nodeID, applyConfig := setupTest(t, ctx, configFn)
+		var nodeMetadata = &structpb.Struct{Fields: map[string]*structpb.Value{
+			"ISTIO_VERSION": {Kind: &structpb.Value_StringValue{StringValue: "1.3"}},
+			"NAMESPACE":     {Kind: &structpb.Value_StringValue{StringValue: nodeID.Metadata.Namespace}},
+		}}
+		listenerReq := &xdsapi.DiscoveryRequest{
+			Node: &xdscore.Node{
+				Id:       nodeID.ServiceNode(),
+				Metadata: nodeMetadata,
+			},
+			TypeUrl: v2.ListenerType,
+		}
+
+		if err := p.StartDiscovery(listenerReq); err != nil {
+			t.Fatal(err)
+		}
+		if err := p.WatchDiscovery(5*time.Second, checkInboundFilterChainTLS); err != nil {
+			t.Error(err)
+		}
+		// apply PeerAuthentication to disable mtls
+		applyConfig()
```

Comment:
The difference for many cases is that we can change the sidecar ingress tls , and i think they are are covered by unit test, is it necessary to test it here again? I think here it is to test the high-level lds generation

Reply:
>It's hard to see how calling this functor disable mTLS. Why don't we just call create and deleteConfig explicit here to be clear?

It is because i donot want to return so many values needed to apply changes. So i wrapped in a function